api/src/legacy_soap_api/legacy_soap_api_proxy.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from os.path import join
+from tempfile import NamedTemporaryFile
+
+from requests import Request, Session
+
+from src.legacy_soap_api.legacy_soap_api_auth import (
+    MTLS_CERT_HEADER_KEY,
+    SessionResumptionAdapter,
+    SOAPClientCertificateLookupError,
+    SOAPClientCertificateNotConfigured,
+)
+from src.legacy_soap_api.legacy_soap_api_config import get_soap_config
+from src.legacy_soap_api.legacy_soap_api_schemas import SOAPRequest, SOAPResponse
+from src.legacy_soap_api.legacy_soap_api_utils import (
+    filter_headers,
+    get_soap_error_response,
+    get_streamed_soap_response,
+)
+
+logger = logging.getLogger(__name__)
+
+# Default proxy request timeout to 1hr
+PROXY_TIMEOUT = 3600
+
+
+def get_proxy_response(soap_request: SOAPRequest, timeout: int = PROXY_TIMEOUT) -> SOAPResponse:
+    config = get_soap_config()
+
+    # Use X-Gg-S2S-Uri header locally if passed, otherwise defualt to GRANTS_GOV_URI:GRANTS_GOV_PORT.
+    proxy_url = join(
+        soap_request.headers.get(config.gg_s2s_proxy_header_key, config.gg_url),
+        soap_request.full_path.lstrip("/"),
+    )
+
+    # Exclude header keys that are utilized only in simpler soap api. Not needed for proxy request.
+    proxy_headers = filter_headers(
+        soap_request.headers, [config.gg_s2s_proxy_header_key, MTLS_CERT_HEADER_KEY]
+    )
+
+    _request = Request(method="POST", url=proxy_url, headers=proxy_headers, data=soap_request.data)
+
+    if not soap_request.auth or config.soap_auth_map == {}:
+        logger.info("soap_client_certificate: Sending soap request without client certificate")
+        return _get_soap_response(_request, timeout=timeout)
+
+    logger.info("soap_client_certificate: Sending soap request with client certificate")
+    # Handle cert based proxy request.
+    with NamedTemporaryFile(mode="w", delete=True) as temp_cert_file:
+        temp_file_path = temp_cert_file.name
+        try:
+            cert = soap_request.auth.certificate.get_pem(config.soap_auth_map)
+        except SOAPClientCertificateLookupError as e:
+            # This exception handles invalid client certs. We will continue to return the response
+            # from GG.
+            cert = ""
+            logger.info(f"soap_client_certificate: Unknown or invalid client certificate: {e}")
+        except SOAPClientCertificateNotConfigured as e:
+            # This exception handles the case of a valid cert being passed, but not configured
+            # to use Simpler SOAP API.
+            logger.info(f"soap_client_certicate: Certificate validated but not configured: {e}")
```

Comment:
Idea: If we wanted to have a good way to create graphs/separate out logs for when these happen, we could add some consistent value to the extra object of these logs.

Something like:
```py
logger.info("whatever message", extra={"soap_error": "soap_client_certificate_lookup"})
```
Then we could easily see how often each of the different error cases happen on a graph that just facets by `soap_error`.